File: 04-quick-sort/ruby/recursive_count_test.rb
# frozen_string_literal: true

require 'minitest/autorun'
require_relative './recursive_count'

class RecursiveCountTest < Minitest::Test
  def test_recursive_count_returns_an_integer
    arr = [1, 2, 3, 4]
    assert_instance_of Integer, recursive_count(arr)
  end

  def test_recursive_count_returns_array_length
    arr = [1, 2, 3, 4]
    assert_equal 4, recursive_count(arr)
  end

  def test_recursive_count_returns_zero_if_list_is_empty
    assert_equal 0, recursive_count([])
  end
end

File: 04-quick-sort/python/recursive_count_test.py
import unittest
from .recursive_count import recursive_count

class TestRecursiveCount(unittest.TestCase):
    def setUp(self):
        pass

    def test_recursive_count_returns_a_number(self):
        arr = [1, 2, 3, 4]
        self.assertIsInstance(recursive_count(arr), int)

    def test_recursive_count_returns_array_length(self):
        arr = [1, 2, 3, 4]
        self.assertEqual(recursive_count(arr), 4)

    def test_recursive_count_returns_zero_if_list_is_empty(self):
        self.assertEqual(recursive_count([]), 0)
